# **Molecular Phylogenetics Course**

By Systematic Biology Group, Lund University

## **Introduction**

In this intensive course we will introduce the basics of phylogenetics, covering both theory and practice. The idea behind the course is that the student gets an overview of the actual methods and learns through hands on exercises to use some of the most used computer programs to study the phylogeny of any group using genetic information.

## **Program**

**Location:** D227 (Retina), Biology Building, SÃ¶lvegatan 35, 223 62 Lund


| Hours\Days | 15th | 16th | 17th |
| ---------- | ---- | ---- | ---- |
| 9:00-10:30 | Lecture 1: Introduction (NW) | Lecture 3: Introduction to model based methods (JR) | Lecture 5: Properties of DNA data and assessing robustness of hypotheses (JR) |
| 10:30-11:00 | Fika | Fika | Fika |
| 11:00-12:30 | Lecture 2: Understanding trees and alignments (NW) | Lecture 4: Statistical frameworks for modelling in phylogenetics (ML and Bayesian) (JR) | Lecture 6: Timing of divergence analyses and using trees for evolutionary studies (NW) |
| 12:30-13:30 | Lunch | Lunch | Lunch |
| 13:30-15:00 | Tutorials 1 and 2 | Tutorials 3, 4, and 5 | Tutorial 6 |
| 15:00-15:30 | Fika | Fika | Fika |
| 15:30-16:30 | Tutorials 1 and 2 | Tutorials 3, 4, and 5 | Tutorial 6 |




## **Index**

### [Course slides](../../tree/master/Lectures)

Here we will upload all the files and information shared during the lectures!


### [Data used in the course](../../tree/master/Data/)

Here you will find the files we create during the course. (If at some point you are at a loss because your attempts to produce the results for the next step, here you can find the files you will need for next steps so you can keep up with the class.)


### [Tutorials](../../tree/master/Tutorials/)

Here are deposited the instructions for the assignments and exercises.


 * [Learn to generate datasets (1)](../../tree/master/Tutorials/1.DatasetManipulation/)
	
 * [Align the gene markers (2)](../../tree/master/Tutorials/2.Alignments/)
 
 * [Select the best evolutionary model and partitioning scheme (3)](../../tree/master/Tutorials/3.ModelSelection/)
 
 * [Maximum likelihood methods (4)](../../tree/master/Tutorials/4.MaximumLikelihood/)
 
 * [Bayesian methods (5)](../../tree/master/Tutorials/5.BayesianInference/)
 
 * [Inference of Times of Divergence (6)](../../tree/master/Tutorials//6.DivergenceTime/)
 

### [Programs used in the tutorials](../../tree/master/Software/)

Here you can find a list of the programs we are going to use and links to where you can download them.
 
 * **Text Editors:**
   	- **TextWrangler** for Mac users
   	- **Notepad++** for Windows users
   	- **Linux** users not covered

 * [MAFFT](https://mafft.cbrc.jp/alignment/software/)

 * [AliView](http://www.ormbunkar.se/aliview/downloads/)

 * [PartitionFinder2](http://www.robertlanfear.com/partitionfinder/)

 * [RAxML](https://antonellilab.github.io/raxmlGUI/)

 * [IQ-TREE](http://www.iqtree.org/)
 
 * [MrBayes](http://nbisweden.github.io/MrBayes/)
	
 * [FigTree](http://tree.bio.ed.ac.uk/software/figtree/)

 * [BEAST2](https://www.beast2.org/)

 * [Tracer](https://github.com/beast-dev/tracer/releases/tag/v1.7.1)

